src/main/scala/ch/epfl/bluebrain/nexus/admin/projects/Projects.scala

```diff
@@ -63,19 +65,18 @@ class Projects[F[_]](agg: Agg[F], index: ProjectCache[F], organizations: Organiz
         index.getBy(organization, label).flatMap {
           case None =>
             val projectId = UUID.randomUUID
-            val base      = project.base.getOrElse(url"${http.prefixIri.asUri}/resources/$organization/$label/_/".value)
-            val vocab     = project.vocab.getOrElse(url"${http.prefixIri.asUri}/vocabs/$organization/$label/".value)
-            val command = CreateProject(projectId,
-                                        label,
-                                        org.uuid,
-                                        organization,
-                                        project.description,
-                                        project.apiMappings,
-                                        base,
-                                        vocab,
-                                        clock.instant,
-                                        caller)
-            evaluateAndUpdateIndex(command) <* retryStrategy(setOwnerPermissions(organization, label, caller))
+            project.baseOrGenerated(organization, label) -> project.vocabOrGenerated(organization, label) match {
```

Comment:
Why use `->` here? it's not a map so it's a bit confusing, perhaps tuple would be more clear.

Reply:
it is, but for me `->` usually means that there's some sort of mapping between elements :)